adapters/adelement/adelement.go

```diff
@@ -0,0 +1,141 @@
+package adelement
+
+import (
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+// Builder builds a new instance of the Adelement adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: template,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+
+	adelementExt, err := getImpressionExt(&request.Imp[0])
+	if err != nil {
+		return nil, append(errors, err)
```

Comment:
I would suggest you to be consistent when you are returning errors. On line 43 - you are returning `[]error{err}`, but here you are appending err to errors var. Requesting you to please change this to `[]error{err}` and remove the declaration of `var errors []error`.

Reply:
I have updated code as per your suggestion.